src/include/btree.i

```diff
@@ -981,8 +981,7 @@ __wt_page_can_evict(WT_SESSION_IMPL *session, WT_PAGE *page, int check_splits)
 	 * been written in the checkpoint, leaving the checkpoint inconsistent.
 	 */
 	if (btree->checkpointing &&
-	    (__wt_page_is_modified(page) ||
-	    F_ISSET(mod, WT_PM_REC_MULTIBLOCK))) {
+	    (__wt_page_is_modified(page) || F_ISSET(mod, WT_PM_REC_MASK))) {
```

Comment:
> `split-insert` only fires for dirty pages, and we prevent eviction of dirty pages during checkpoints. All of the checks are in `__wt_page_can_evict`.

Fair enough -- my point is that `__wt_split_insert` itself checks for a dirty page, I had to wander around to decide it wasn't a problem.